packages/devtools_app/lib/src/shared/table.dart

```diff
@@ -845,34 +843,28 @@ class _TableState<T> extends State<_Table<T>> with AutoDisposeMixin {
     sortColumn = widget.sortColumn;
     sortDirection = widget.sortDirection;
     scrollController = ScrollController();
-
+    _addScrollListener(widget.selectionNotifier);
     _initSearchListener();
   }
 
-  @override
-  void didUpdateWidget(_Table<T> oldWidget) {
-    super.didUpdateWidget(oldWidget);
 
-    cancelListeners();
+  void _addScrollListener(ValueNotifier<Selection<T>>? selectionNotifier) {
+    if (selectionNotifier != null) {
+      addAutoDisposeListener(selectionNotifier, () {
+        WidgetsBinding.instance.addPostFrameCallback((_) {
+          final Selection<T> selection = selectionNotifier.value;
+          if (selection.scrollIntoView) {
+            final int selectedDisplayRow = selection.nodeIndexCalculator != null
+                ? selection.nodeIndexCalculator!(selection.node!)
+                : selection.nodeIndex!;
```

Comment:
Instead of null-asserting everywhere, let's pull these out into some variables, and only scroll if we have a node and a way of getting the node index:

```
if (scrollIntoView && node != null && (nodeIndexCalculator != null || nodeIndex != null)) {
}
```